lib/modules/datasource/maven/index.spec.ts

```diff
@@ -189,6 +205,31 @@ describe('modules/datasource/maven/index', () => {
     expect(res).toMatchSnapshot();
   });
 
+  it('returns releases when only snapshot', async () => {
+    const meta = loadFixture('metadata-snapshot-version.xml');
+    mockGenericPackage({
+      meta: loadFixture('metadata-snapshot-only.xml'),
+      jars: null,
+      html: null,
+      latest: '1.0.3-SNAPSHOT',
+      snapshots: [
+        {
+          version: '1.0.3-SNAPSHOT',
+          meta: meta,
+          jarStatus: 200,
+        },
+      ],
+    });
+    httpMock
+      .scope(baseUrl)
+      .get('/org/example/package/1.0.3-SNAPSHOT/maven-metadata.xml')
+      .reply(200, meta);
+
+    const res = await get();
+
+    expect(res).toMatchSnapshot();
```

Comment:
```suggestion
    expect(res).toMatchObject({
      group: 'org.example',
      name: 'package',
      releases: [{ version: '1.0.3-SNAPSHOT' }],
    });
```

Reply:
What's that advantage of using `toMatchObject` instead of `toMatchSnapshot`?

They both seem to accomplish the same goal.

Reply:
The advantage is that `toMatchObject` operates in-line and is way easier to analyze for future reader of your test. Also, `toMatchSnapshot()` has a bigger chance to be accidentally re-written with the wrong data (when updating snaphots in bulk with other tests). That incorrect value further will be considered as the correct one for indefinite long period of time.